packages/styled/src/base.js

```diff
@@ -196,12 +195,16 @@ let createStyled: CreateStyled = (tag: any, options?: StyledOptions) => {
       nextTag: ElementType,
       nextOptions?: StyledOptions
     ) => {
-      return createStyled(
-        nextTag,
-        nextOptions !== undefined
-          ? { ...(options || {}), ...nextOptions }
-          : options
-      )(...styles)
+      return createStyled(nextTag, {
+        ...options,
+        ...nextOptions,
+        shouldForwardProp:
+          nextOptions && typeof nextOptions.shouldForwardProp === 'function'
+            ? propName =>
+                nextOptions.shouldForwardProp(propName) &&
+                defaultShouldForwardProp(propName)
+            : defaultShouldForwardProp
```

Comment:
My understanding here is that we want to pipe the should forward props like a funnel (C -> B -> A) using either the supplied `shouldForwardProp` falling back to the composed `defaultShouldForwardProp` of the parent.